### DISCUSSION TITLE
Debounce useAsyncData that uses a watcher

### DISCUSSION THREAD

**Discussion by qnp** [nuxt/nuxt#16337] [Ideas]
Hi there,

When using a `useAsyncData` with a `watch` option, I wonder if there is a clean way to debounce that watcher ? The use case is an autocomplete which calls an API each time the input changes, but we want to prevent too swift successive requests while the user types in.

I see that VueUse has a [refDebounced](https://vueuse.org/shared/refDebounced/) implementation, so I can simply use a `refDebounced` on my watcher instead of a regular `ref`. Yet, in other cases, a `debounce` option on useAsyncData might be useful, for instance if `refresh` is programmatically called elsewhere very quickly with some other mechanics.

**Comment by am05mhz** [comment]
i need this option too for the `watch` parameter

as for the `refresh` function, according to the docs, it has a parameter in which you can pass `{dedupe: true}` eg: `refresh({dedupe: true})`
as described in the `RefreshOptions` in https://nuxt.com/docs/api/composables/use-async-data, but there is no mention of what it do whatsoever.
also currently i prefer the `watch` way over `refresh`, so i have not tried it yet

**Comment by localusercamp** [comment]
`Lodash` and `VueUse` already has utilities to solve this.
I think 99% of all projects that written on nuxt must have one of the two nowadays.

**Reply by IlyaSemenov** [reply]
They are not directly suitable. Using `refDebounced` leaves a time window when the value has already changed but not yet debounced. `useAsyncData` result should be marked as pending during that time, which is not happening. This should somehow go into the core.
